blockchain/validators/src/avr_validator.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Types for reading and writing historical Intel Attestation
+//! Verification Reports from MobileCoin consensus enclaves
+
+use displaydoc::Display;
+use mc_attest_core::VerificationReport;
+use mc_blockchain_types::BlockIndex;
+use mc_common::ResponderId;
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, DisplayFromStr};
+use std::collections::BTreeMap;
+
+/// Object representing AVR history for all consensus nodes
+#[serde_as]
+#[derive(Clone, Debug, Display, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub struct AvrRecords {
+    /// AVR Records
+    #[serde_as(as = "BTreeMap<_, BTreeMap<DisplayFromStr, _>>")]
+    pub avr_records: BTreeMap<ResponderId, BTreeMap<BlockIndex, AvrRecord>>,
```

Comment:
Consider using an intermediate type for serde, which more closely matches the config file structure, and then this class can take that config object and map that to its internal representation, or the config struct can map itself into something you can pass to `AvrValidator::new()`. That's the approach I took in #2131, specifically https://github.com/mobilecoinfoundation/mobilecoin/blob/metadata-signers/blockchain/validators/src/metadata/key_range/config.rs